CMakeLists.txt

```diff
@@ -67,6 +67,12 @@ set(
   src/event/touch.cpp
   )
 
+#### Set compilation flags {{{
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11")
```

Comment:
about target_compile_definitions/options, it's because it avoid issues with polluting the context with variable which might be taken into account where they should not.

For the error, I'll see soon if it can be avoided, I'm a bit surprised.